### PR TITLE
Make confluenceUpdatePagesParentIdsActivity more efficient

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

We encountered issues with large Confluence spaces where calculating the full space hierarchy to update our tags in the vector database exceeded our temporal timeout. This PR introduces parallel processing to accelerate the process.

### CODE REVIEW

**Review by philipperolet — COMMENTED** [review]
Just checking: did you ensure that updateDocumentParentsField was "thread-safe" ?

**Comment by flvndvd** [comment]
> updateDocumentParentsField

It should, it only performs http call on core's API.

**Review by philipperolet — APPROVED** [review]
:+1: thanks a lot :pray:

File: connectors/src/connectors/confluence/temporal/activities.ts [hunk 1]
```diff
@@ -473,25 +474,30 @@ export async function confluenceUpdatePagesParentIdsActivity(
...
-      documentId: makeConfluencePageId(page.pageId),
-      parents: parentIds,
-    });
-  }
+  await concurrentExecutor(
+    pages,
+    async (page) => {
+      // Retrieve parents using the internal ID, which aligns with the permissions
+      // view rendering and RAG requirements.
+      const parentIds = await getConfluencePageParentIds(
```
Comment (flvndvd) [inline connectors/src/connectors/confluence/temporal/activities.ts:483]: I'll do a follow up PR to change the types here to use a readonly map to make it more explicit that this is thread safe.

**Pull request merged**